message-markdown.md

```diff
@@ -0,0 +1,82 @@
+# Markdown in messages
+> This list is for every formating thing that goes beyond plain-text.
+
+Deltachat is going to support the following subset of markdown:
+
+### `*italics*` and `_italics_`
+
+### `**bold**` and `__bold__`
+
+### `~~strikethrough~~`
+
+### `https://delta.chat` - Urls
+
+Make URLs clickable, but **don't** support the Link syntax `[Name](url)`
+because that could be used for misleading and as such be used in scams.
+
+### `<http://example.org>`  - Urls
+
+### `` `inline-code` ``
+useful to send non markdown text in your message like source code snippets.
+Should get rendered in a monospace font and with a different background.
+ 
+### ` ``` fence code block ``` `
+
+```
+Similar to `inline-code` but not inline and it may supports code highlighting.
+```
+` ```[lang?] [content]``` `
+ A bit modified from the common syntax to allow one liners.
+Also get displayed with an monospace font (a side effect of this is that it allows to display small ascii art).
+The code **highlighting** is **optional** as implementation (time)cost 
+may not be worth the small gain.
+The `language` definition should be parsed separately and omitted in this case.
+
+see https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#code-and-syntax-highlighting
+
+### `:emoji:`
+- could also be used for custom dc emojis
+
+### labeled links: `[Name](url)` links
+When implementing this make sure to show the user the hidden url in a confirmation dialog to make scamming harder.
+Also show the url as encode punycode to make punycode attacks useless.
+Optionaly a client can implement a system to trust an domain (an "don't ask a again for links on this domain" checkbox in the confirmation dialog)
+
+### Bot `/commands`
```

Comment:
Regex could look like this (but keep in mind that this doesn't allow whitespaces inside of the command):
```js
/(\/[a-Z0-9]+)\s/
```

Reply:
also should the current draft be replaced by it or only if the draft is empty? CC @r10s @adbenitez

Reply:
the current android implementation taken from @adbenitez replaces the draft.